src/ape/api/providers.py

```diff
@@ -179,8 +179,12 @@ class ProviderAPI(BaseInterfaceModel):
     plugin or the `ape-hardhat <https://github.com/ApeWorX/ape-hardhat>`__ plugin.
     """
 
+    # TODO: In 0.9, make not optional.
```

Comment:
an alt to this would be to hav `@property name(self)` which by default returns the `__class__.name__`

Reply:
little tricky as it needs to be known prior to instantiation